alana/prompt.py

```diff
@@ -257,7 +273,7 @@ def gen_examples(instruction: str, n_examples: int = 5, model: str = globals.DEF
     formatted_examples: str = "\n<examples>\n<example>" + '</example>\n<example>'.join(examples) + "</example>\n</examples>"
     return formatted_examples
 
-def gen_prompt(instruction: str, messages: Optional[List[MessageParam]] = None, model: str = globals.DEFAULT_MODEL, api_key: Optional[str] = None, max_tokens: int = 1024, temperature=1.0, **kwargs: Any) -> Dict[Literal["system", "user", "full"], Union[str, List]]:
+def gen_prompt(instruction: str, messages: Optional[List[MessageParam]] = None, model: str = globals.DEFAULT_MODEL, api_key: Optional[str] = None, max_tokens: int = 1024, temperature=1.0, **kwargs: Unpack[RequestParams]) -> Dict[Literal["system", "user", "full"], Union[str, List]]:
```

Comment:
This is a bit problematic, because I also want to be able to pass kwargs into `gen` that aren't accepted by the Claude API (e.g. `loud`).

Can we maybe add a kwargs param to `RequestParams`?

Reply:
You can give me the arguments with there types and I can update the class. If the code will grow and additional arguments will be needed then the class my be updated.

Addition arguments either way will be possible to pass, they just will not be type checked if not specified in this class.